File: src/main/kotlin/com/stannis/dataModel/complexStatementTypes/ComplexCompositeTypeSpecifier.kt
package com.stannis.dataModel.complexStatementTypes

import com.stannis.dataModel.Statement
import com.stannis.dataModel.statementTypes.CompositeTypeSpecifier
import com.stannis.dataModel.statementTypes.Name

data class ComplexCompositeTypeSpecifier(
    override val `$type`: String? ="ComplexCompositeTypeSpecifier",
    var our_class: CompositeTypeSpecifier,
    var path: String,
    var library: List<Name>
): Statement {
    fun clone(): ComplexCompositeTypeSpecifier {
         return ComplexCompositeTypeSpecifier(this.`$type`, this.our_class, this.path, this.library)
    }
}

File: Interpreter/Interpreter/Models/serialize/complexStatementTypes/ComplexCompositeTypeSpecifier.cs
using System.Collections.Generic;
using Interpreter.Models.serialize.statementTypes;

namespace Interpreter.Models.serialize.complexStatementTypes{
    public class ComplexCompositeTypeSpecifier: IStatement
    {
        public CompositeTypeSpecifier our_class;

        public string path;

        public IList<IStatement> library = new List<IStatement>();
    
    }
}